### PR TITLE
Implement Onboarding screens and QRCode reading

### PR DESCRIPTION
**Pull request by wbenabdessalem**
FT Android: Reach the level of iOS App

### CODE REVIEW

File: app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java [hunk 1]
```diff
@@ -41,23 +43,48 @@
...
+    // Read previous value. If not found, use 0 as default value.
+    try {
+      rootURL = new URL(remoteMessage.getData().get("url"));
+      System.out.println("rootURL ===> " + rootURL.toString());
+      String urlDomain = rootURL.getHost();
+      String urlProtocol = rootURL.getProtocol();
+      String urlKey = urlProtocol + "://" + urlDomain;
+      System.out.println("urlProtocol ===> " + urlProtocol);
+      System.out.println("getHost ===> " + urlDomain);
+      System.out.println("urlKey ===> " + urlKey);
```
Comment (ahamdi) [inline app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java]: To remove

File: app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java [hunk 2]
```diff
@@ -41,23 +43,48 @@
...
+      count = shared.getInt(urlKey, 0);
+      count = count + 1;
+      // we need to save the value again for another badge count
+      SharedPreferences.Editor editor = shared.edit();
+      editor.putInt(urlKey, count);
+      editor.apply();
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
```
Comment (ahamdi) [inline app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java]: Use logger 
```suggestion
      Log.e("Malformed URL",e);
```

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java (hunk 2 above)
Comment (wbenabdessalem) [reply app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java]: Done

File: app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java [hunk 3]
```diff
@@ -41,23 +43,45 @@
...
     super.onMessageReceived(remoteMessage);
     Log.d(TAG, "onMessageReceived: " + remoteMessage.getFrom());
     Log.d(TAG, "onMessageReceived: " + remoteMessage.getData());
-
+//import android.preference.PreferenceManager;
+    SharedPreferences shared = PreferenceManager.getDefaultSharedPreferences(this);
+    // Read previous value. If not found, use 0 as default value.
+    try {
+      rootURL = new URL(remoteMessage.getData().get("url"));
+      System.out.println("rootURL ===> " + rootURL.toString());
```
Comment (ahamdi) [inline app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java:72]: to remove

File: app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java [hunk 4]
```diff
@@ -41,23 +43,45 @@
...
+      String urlDomain = rootURL.getHost();
+      String urlProtocol = rootURL.getProtocol();
+      String urlKey = urlProtocol + "://" + urlDomain;
+      count = shared.getInt(urlKey, 0);
+      count = count + 1;
```
Comment (ahamdi) [inline app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java:77]:
```suggestion
      count = shared.getInt(urlKey, 0) + 1;
```

File: app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java [hunk 5]
```diff
@@ -41,23 +43,45 @@
...
-
+//import android.preference.PreferenceManager;
+    SharedPreferences shared = PreferenceManager.getDefaultSharedPreferences(this);
+    // Read previous value. If not found, use 0 as default value.
+    try {
+      rootURL = new URL(remoteMessage.getData().get("url"));
+      System.out.println("rootURL ===> " + rootURL.toString());
+      String urlDomain = rootURL.getHost();
+      String urlProtocol = rootURL.getProtocol();
+      String urlKey = urlProtocol + "://" + urlDomain;
```
Comment (ahamdi) [inline app/src/main/java/org/exoplatform/service/push/PushNotificationsService.java:75]: Get the URL with one instruction

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 6]
```diff
@@ -14,11 +14,11 @@ android {
         applicationId "org.exoplatform"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.compileSdkVersion
-        versionCode 01
-        versionName "6.2.0"
+        versionCode 43
+        versionName "5.1.7"
```
Comment (ahamdi) [inline app/build.gradle:18]:
```suggestion
        versionCode 01
        versionName "6.2.0"
```

File: app/build.gradle [hunk 7]
```diff
@@ -49,12 +49,6 @@ android {
             versionNameSuffix "-beta"
             manifestPlaceholders = [appLabelEnvironmentSuffix:" (beta)"]
         }
-        dev {
-            dimension "mode"
-            applicationIdSuffix ".dev"
-            versionNameSuffix "-dev"
-            manifestPlaceholders = [appLabelEnvironmentSuffix:" (dev)"]
-        }
```
Comment (ahamdi) [inline app/build.gradle:57]: To restore

File: app/build.gradle [hunk 8]
```diff
@@ -63,7 +57,7 @@ android {
     variantFilter { variant ->
         def names = variant.flavors*.name
         if (variant.buildType.name.contains("debug")) {
-            if (names.contains("dev") || names.contains("beta") || names.contains("store")) {
```
Comment (ahamdi) [inline app/build.gradle:66]: To restore

**Pull request closed without merging**